File: src/tasks/task9/Solution.java
package tasks.task9;

import java.util.Stack;

import static java.lang.Integer.parseInt;
import static java.lang.String.valueOf;

/**
 * https://app.codility.com/c/feedback/QY5MEQ-K6Z/
 */

public class Solution {

    public int solution(String S) {

        try {

            Stack<Integer> stack = new Stack<>();

            for (char s : S.toCharArray())
                if (s == '+')
                    stack.push(stack.pop() + stack.pop());
                else if (s == '*')
                    stack.push(stack.pop() * stack.pop());
                else
                    stack.push(parseInt(valueOf(s)));

            return stack.pop();

        } catch (Exception e) {

            return -1;
        }
    }

    public static void main(String[] args) {

        Solution solution = new Solution();

        System.out.println(solution.solution("13+62*7+*"));
        System.out.println();

        System.out.println(solution.solution("11++"));
        System.out.println();
    }
}

File: src/tasks/task9/solution.py
import sys

'''
    https://app.codility.com/c/feedback/NDJ749-X8Y/
'''

def solution(S):
    try:

        stack = []
        for s in S:
            if s == '+':
                stack.append(stack.pop() + stack.pop())
            elif s == '*':
                stack.append(stack.pop() * stack.pop())
            else:
                stack.append(int(s))

        return stack.pop()

    except:
        return -1



def main(argv):
    print(solution("13+62*7+*"))
    print()

    print(solution("11++"))
    print()



if __name__ == "__main__":
    main(sys.argv[1:])
